apps/studio/components/layouts/TableEditorLayout/TableEditorMenu.tsx

```diff
@@ -70,7 +70,7 @@ const TableEditorMenu = () => {
       sort,
     },
     {
-      keepPreviousData: true,
+      keepPreviousData: Boolean(searchText),
```

Comment:
This is the main change. The other two instances of `keepPreviousData` were removed because they weren't really doing anything.